src/Azure.Functions.Cli/Helpers/PythonHelpers.cs

```diff
@@ -161,7 +161,9 @@ internal static async Task<Stream> GetPythonDeploymentPackage(IEnumerable<string
         private static async Task<Stream> InternalPreparePythonDeployment(IEnumerable<string> files, string functionAppRoot)
         {
             var packagesPath = await RestorePythonRequirements(functionAppRoot);
-            return ZipHelper.CreateZip(files.Concat(FileSystemHelpers.GetFiles(packagesPath)), functionAppRoot);
+            Stream zipStream = ZipHelper.CreateZip(files.Concat(FileSystemHelpers.GetFiles(packagesPath)), functionAppRoot);
+            FileSystemHelpers.DeleteDirectorySafe(packagesPath);
```

Comment:
I'm not sure this would work. I thought the zip stream is a lazy stream. I think it'd be safer to delete that folder in `RestorePythonRequirements` if it already exists before creating it

Reply:
I was initially wondering the same thing. But, I actually tested it multiple times and it seems to always work fine, so I went ahead with it. Maybe it's because how we are using a memorystream and copying it to the zip stream for each file that makes it store in the memory?

But, if you would rather want me to be safe, I can remove it elsewhere. I would say it might be better to do it in `GetPythonDeploymentPackage`. Because if the user did `publish --build-native-deps`, we don't go through `RestorePythonRequirements`. 
So, if the user ran `func pack` and then `func pack --build-native-deps`, they would have the large .`python_packages` in the zip file for no reason.

Reply:
I see. I think you might be right about loading all the files in a MemoryStream because I think that's the underlying issue of this https://github.com/Azure/azure-functions-core-tools/issues/696 

I'm fine with whatever makes sense to you

Reply:
On a second thought, is there a reason why we create `.python_packages` directory in the functionapp directory over doing so in a temp location, similar to what we do for `--build-native-deps`?

Reply:
I see. For now,  I just updated the PR to delete if it's present anytime `pack` or `publish` is called on a python project.

That should cover most of the scenarios.